web/src/app/lists/FlatList.js

```diff
@@ -90,21 +97,31 @@ export default class FlatList extends React.PureComponent {
         button: true,
       }
     }
+
     return (
       <ListItem
         key={idx}
         {...itemProps}
         style={{ width: '100%' }}
         className={item.highlight ? this.props.classes.highlightedItem : null}
       >
-        {item.icon}
+        {item.icon && <ListItemIcon>{item.icon}</ListItemIcon>}
         <ListItemText
           primary={item.title}
           secondary={item.subText}
           secondaryTypographyProps={{ style: { whiteSpace: 'pre-line' } }}
         />
-        {item.action && (
-          <ListItemSecondaryAction>{item.action}</ListItemSecondaryAction>
+        {(item.secondaryAction || item.tertiaryAction) && (
+          <ListItemSecondaryAction>
+            <Grid
+              container
+              spacing={2}
+              className={this.props.classes.actionGrid}
+            >
+              {item.tertiaryAction && <Grid item>{item.tertiaryAction}</Grid>}
```

Comment:
So this is my bad, but I didn't realize these were both being rendered as the `ListItemSecondaryAction` in the same place when I suggested using `secondaryAction` and `tertiaryAction`.

If these are always rendered in the same place, I don't think we actually need two and can just use `action` as was previous.

We can use the original `<ListItemSecondaryAction>{item.action}</ListItemSecondaryAction>` code here and then in the one place we do something different, just pass in <Grid> and it's children as the `action` prop and we'll get the same result without introducing additional props.

Reply:
Oh that's a good idea. So just leave action as is, but have it be a grid that we pass from the parent.  🎉 for collaboration!